packages/pipeline-editor/src/RuntimesWidget.tsx

```diff
@@ -28,12 +28,29 @@ import React from 'react';
 
 import { PipelineService, RUNTIMES_NAMESPACE } from './PipelineService';
 
+const RUNTIMES_METADATA_CLASS = 'elyra-metadata-runtimes';
+
+const addTrailingSlash = (url: string): string => {
+  return url.endsWith('/') ? url : url + '/';
+};
+
+const getGithubURLFromAPI = (apiEndpoint: string): string => {
+  // For Enterprise Server the api is located at <hostname>/api/
+  let baseURL = new URL(apiEndpoint).origin;
+
+  // For Github.com and Github AE the api is located at api.<hostname>
+  baseURL = baseURL.replace('api.', '');
+
+  return addTrailingSlash(baseURL);
```

Comment:
URL (from nodejs) has the option to handle join. Could we use that instead of implementing our own code for that?

Reply:
We could but it wouldn't really change the code, `addTrailingSlash` conditionally adds `/` if it's not already there. So if we used `URL.join` we would need to strip the `/` off first. In the end the amount of code wouldn't change, it would just be different. I'm ok either way, I did consider it, but didn't bother when I realized it wouldn't think the code footprint

Reply:
The [docs](https://nodejs.org/api/url.html#url_the_whatwg_url_api) seem to imply no striping is needed.

const myURL = new URL('/foo', 'https://example.org/');
// https://example.org/foo

Reply:
Copying what I mentioned in a Webex here: in general trailing slash is not an issue, but we intentionally added it in our code to fix an issue related to kfp instances. I've just updated our code to add it for all our links here